packages/cli/src/code-format.ts

```diff
@@ -0,0 +1,187 @@
+        type: 'number',
+        description: 'Indentation size',
+        alias: 's',
+        default: 4,
+    })
+    .option('indentWithTabs', {
+        type: 'boolean',
+        description: 'Indent with tabs, overrides -s and -c',
+        alias: 't',
+        default: false,
+    })
+    .option('maxPerserveNewlines', {
+        type: 'number',
+        description: 'Maximum number of line-breaks to be preserved in one chunk',
+        alias: 'm',
+        default: 1,
+    })
+    .option('newlineBetweenRules', {
+        type: 'boolean',
+        description: 'Add a newline between CSS rules',
+        alias: 'N',
+        default: true,
+    })
+    .option('perserveNewlines', {
+        type: 'boolean',
+        description: 'Preserve existing line-breaks',
+        alias: 'p',
+        default: true,
+    })
+    .option('selectorSeparatorNewline', {
+        type: 'boolean',
+        description: 'Add a newline between multiple selectors',
+        alias: 'L',
+        default: true,
+    })
+    .option('debug', {
+        type: 'boolean',
+        description: 'Enable explicit debug log',
+        alias: 'D',
+        default: false,
+    })
+    .option('require', {
+        type: 'array',
+        description: 'require hooks',
+        alias: 'r',
+        default: [] as string[],
+    })
+
+    .alias('h', 'help')
+    .alias('v', 'version')
+    .help()
+    .strict()
+    .wrap(yargs.terminalWidth())
+    .parseSync();
+
+const {
+    debug,
+    endWithNewline,
+    indentEmptyLines,
+    indentSize,
+    indentWithTabs,
+    maxPerserveNewlines,
+    newlineBetweenRules,
+    perserveNewlines,
+    require: requires,
+    selectorSeparatorNewline,
+    target,
+} = argv;
+
+const log = createLogger('[Stylable code formatter]', debug);
+
+log('[Arguments]', argv);
+
+// execute all require hooks before running the CLI build
+for (const request of requires) {
+    if (request) {
+        require(request);
+    }
+}
+
+const stylable = Stylable.create({
+    fileSystem: nodeFs,
+    requireModule: require,
+    projectRoot: target,
+    resolverCache: new Map(),
+});
+const lsp = new StylableLanguageService({ fs: nodeFs, stylable });
+
+function readDirectoryDeep(dirPath: string, fileSuffixFilter = '.st.css') {
+    const files = nodeFs.readdirSync(dirPath, 'utf-8');
+    let res: string[] = [];
+
+    for (const item of files) {
+        const currentlFilePath = join(dirPath, item);
+        const itemStat = nodeFs.statSync(join(currentlFilePath));
+
+        if (itemStat.isFile() && item.endsWith(fileSuffixFilter)) {
+            res.push(currentlFilePath);
+        } else if (itemStat.isDirectory()) {
+            res = res.concat(readDirectoryDeep(currentlFilePath));
+        }
+    }
+
+    return res;
+}
+
+function formatStylesheet(filePath: string) {
+    log('Formatting: ' + filePath);
+    const fileContent = nodeFs.readFileSync(filePath, 'utf-8');
+
+    const formatting = lsp.formatDocument(
```

Comment:
we need to see if we can not use the `formatDocument` from lsp because it creates bloat. if possible move the function somewhere else and reuse it in lsp.

Reply:
We can probably do something like that, not sure if it should block this from going in

Reply:
I'm doing it.

Reply:
Done.